### ISSUE TITLE
Parcel2 Unable to set breakpoints in browser debugger

### ISSUE THREAD

**Issue by pingxiong620** [parcel-bundler/parcel#8691]
I am upgrading from parcel-bundler to parcel 2. I can't set any breakpoint in browser (chrome) debugger (breakpoint seems to be disabled for all lines) when "sourceMap": true in tsconfig.json, and "--no-optimize" is specified on parcel build command line. My source are .ts and .tsx files, and I can see the source tree on browser. Before upgrading, I was able to debug into each line on a browser. 
parcel version is 2.8.1
.parcelrc contains
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.{ts,tsx}": [
      "@parcel/transformer-typescript-tsc"
    ]
  }
}

When using the default parcel transpiler (remove the .parcelrc file) I could set some breakpoints on some lines however not every line could be hit by the debugger (both Chrome and Firefox).

I saw someone already reported the same issue in this thread, but I don't see any solutions. 
https://github.com/parcel-bundler/parcel/issues/8086

**Comment by Nick-Riggs** [comment]
I am seeing this as well but in plain javascript. I can set breakpoints in DevTools, but they are not mapped to the correct lines of code.

**Comment by mischnic** [comment]
@pingxiong620 @Nick-Riggs Please share a complete reproduction where this happens

**Comment by pingxiong620** [comment]
Initiated a react app here . The app shows the default react application with a button. When click the button it will call a method, set a breakpoint inside the method then step over the statements, some statements could not be hit or set a breakpoint. 
https://github.com/pingxiong620/Parcel2

1. npm install
2. npm run pstart

**Comment by Nick-Riggs** [comment]
Nice work @pingxiong620. I have verified that the example demonstrates the problem for me as well. 

@mischnic, please note that the breakpoint will hit, but if you examine the locals and start stepping, it will become apparent the code is not synced to the breakpoint correctly. 

You can also just add a `debugger;` statement and you'll see where the code breaks, isn't on the statement.

**Comment by comm644** [comment]
Without  'transformer-typescript-tsc' paths - ok

other reproducer:
//index.js
```js

import "./styles.css"
import "./src/other";

console.log("test");
```

//src/other.ts
```js
console.log ("other");
```

//index.html
```html
<html>
<body>
<script src="dist/index.js"></script>
</body>
</html>

```

//.parcelrc
```json
{
  "extends": "@parcel/config-default",
  "transformers": {
    "*.{ts,tsx}": ["@parcel/transformer-typescript-tsc"]
  },
  "namers": ["parcel-namer-hashless"]
}
```

//package.json
```json
{
  "name": "parcel-test",
  "version": "1.0.0",
  "license": "MIT",
  "devDependencies": {
    "@parcel/transformer-typescript-tsc": "^2.8.2",
    "parcel": "^2.8.2",
    "parcel-namer-hashless": "^1.0.3",
    "typescript": ">=3.0.0"
  },
  "babel": {
    "presets": [
      "@babel/preset-react",
      "@babel/preset-flow",
      "@babel/preset-typescript",
      "@babel/preset-env"
    ],
    "plugins": [
      "@babel/plugin-transform-runtime",
      "@babel/plugin-proposal-class-properties",
      "@babel/plugin-transform-flow-strip-types",
      "minify-dead-code-elimination"
    ]
  },
  "targets": {
    "default": {
      "context": "browser",
      "outputFormat": "global",
      "isLibrary": false,
      "sourceMap": {
        "inline": true,
        "inlineSources": true,
        "sourceRoot": "/"
      },
      "publicUrl": "/dist/"
    }
  }
}

```

//tsconfig.json
```json
{
  "compilerOptions": {
    "jsx": "react",
    "allowJs": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true
  }
}
```

**Comment by mischnic** [comment]
@pingxiong620 @Nick-Riggs Perfect, somehow swc produces invalid sourcemaps with block comments before functions: https://github.com/swc-project/swc/issues/6694

**Comment by mischnic** [comment]
That specific case was actually broken due to Windows line endings. @Nick-Riggs Are you on Windows/using Windows line endings as well?

**Comment by pingxiong620** [comment]
May I ask when the new version of parcel with the fix will be available? Thx

**Comment by scott-callisto** [comment]
I am having this exact same problem and I do not have Windows line endings

**Comment by mischnic** [comment]
That windows line ending fix was just released.

@scott-callisto Please share a reproduction in a new issue.

**Comment by scott-callisto** [comment]
Sadly, the only reproduction I can get is in my main application, which is very complex and proprietary to boot. When I try to create a simpler application to share, it all works perfectly. I'll create an issue if/when I get a reproduction that I can actually share.

**Comment by jqweryjson** [comment]
Same issuses, when "sourceMap": true, - it is all broken. 
When "inlineSourceMap": true, i can set breakpoint, but it is wrong line =(

**Comment by mischnic** [comment]
@jqweryjson please share a reproduction

**Comment by jqweryjson** [comment]
> @jqweryjson please share a reproduction

unfortunately i have the same situation as scott-callisto comment above.
Anyway, removing "*.{ts,tsx}": ["@parcel/transformer-typescript-tsc"] from  transformers helped me

**Comment by mischnic** [comment]
And you are using Parcel 2.8.3?

**Comment by scott-callisto** [comment]
I can't speak for @jqweryjson, but I have that problem when using Parcel 2.8.3 (breakpoints go in on the wrong lines). When using Parcel 2.8.1, I can't set breakpoints at all. Removing `"*.{ts,tsx}": ["@parcel/transformer-typescript-tsc"]` from .parcelrc transformers is not an option for me, as we use tsconfig compiler options that require it.

**Comment by jqweryjson** [comment]
My env is

    "@parcel/config-default": "2.5.0",
    "@parcel/core": "2.5.0",
    "@parcel/optimizer-terser": "2.5.0",
    "@parcel/packager-css": "2.5.0",
    "@parcel/transformer-less": "2.5.0",
    "@parcel/transformer-sass": "2.5.0",
    "@parcel/transformer-svg": "2.5.0",
    "@parcel/transformer-svg-react": "2.5.0",
    "@parcel/transformer-typescript-tsc": "2.5.0",
    "@parcel/transformer-yaml": "2.5.0",
    
  This is legacy project.  I can't bump versions.
  First step i added "inlineSourceMap": true in tsconfig. I get working breakpoints, but it`s all stops in a wrong line
  Second, i remove "@parcel/transformer-typescript-tsc" and get all working

**Comment by valepu** [comment]
I'm currently facing the same issue, using parcel 2.8.3. On chrome sometimes breakpoints work (like once every 20-30 reloads), on Mozilla they never work. I'll try to replicate the issue on an empty project

**Comment by pingxiong620** [comment]
We ended up to use webPack.

**Comment by valepu** [comment]
I found out what the issue was for me.
During development i add the current timestamp as a parameter to the `<script>` tag , like this

`<script type="text/javascript" src="index.js?x=1680128304"></script>`

I've been working to this project for months and this never caused an issue, but now all of a sudden it doesn't work. I don't know if it's an issue with Parcel or with browsers. I haven't updated my libraries in the last weeks (I have updated them after I got this issue but it didn't help) so I guess something changed recently with browsers?
